rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+    curvature : dictionary
+    n2d order polynominal coeficients defining
+    curvature
+    {'x^2': coef, 'x1': coef, 'x0': coef}
+
+    Returns
+    -----------
+
+    """
+    x_centers, offsets = get_curvature_offsets(photon_events, bins)
+    poly_model = lmfit.Model(poly, independent_vars=['x'])
+    params = poly_model.make_params()
+    for name, c in guess_curvature.items():
+        params[name].set(value=c)
+    for name, TF in vary_params.items():
+        params[name].set(value=TF)
+    result = poly_model.fit(offsets, x=x_centers, params=params)
+
+    if result.success is not True:
+        print(result.message)
+
+    return result.best_values
+
+
+def plot_curvature(ax1, curvature, photon_events, plot_offset=0.):
+    """ Plot a red line defining curvature on ax1
+
+    Parameters
+    ----------
+    ax1 : matplotlib axes object
+        axes for plotting on
+    curvature : dictionary
+        n2d order polynominal coeficients defining image curvature
+        {'x^2': coef, 'x1': coef, 'x0': coef}
+    photon_events : array
+        two column x, y photon location coordinates
+
+    Returns
+    ---------
+    curvature_artist : matplotlib artist object
+        artist from image scatter plot
+    """
+    x = np.arange(np.nanmax(photon_events[:, 0]))
+    y = poly(x, **curvature)
+    return plt.plot(x, y + plot_offset, 'r-')
+
+
+def estimate_elastic_pos(photon_events, x_range=(0, 20), bins=None):
+    """
+    Estimate where the elastic line.
+
+
+    Parameters
+    ----------
+    photon_events : array
+        two column x, y, Iph photon locations and intensities
+    x_range : (float, float)
+        Select x region of image to use
+        A tuple defines a range in x (minx, maxx)
+    bins : int or sequence of scalars or str, optional
+        Binning in the y direction.
+        If 'bins' is None a step of 1 is assumed over the relavant range
+        If `bins` is an int, it defines the number of equal-width
+        bins in the given range (10, by default). If `bins` is a
+        sequence, it defines the bin edges, including the rightmost
+        edge, allowing for non-uniform bin widths.
+    """
+    x = photon_events[:, 0]
+    y = photon_events[:, 1]
+    Iph = photon_events[:, 2]
+
+    choose = np.logical_and(x > x_range[0], x < x_range[1])
+    y_choose = y[choose]
+    I_choose = Iph[choose]
+
+    if bins is None:
+        bins = np.arange(y_choose.min()//1 + 0.5, y_choose.max()//1 - 0.5)
+    I_profile, y_edges = np.histogram(y_choose, bins=bins, weights=I_choose)
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    elastic_y_value = y_centers[np.argmax(I_profile)]
+    return elastic_y_value
+
+
+def extract(photon_events, curvature, bins=None):
```

Comment:
I am not sure that 'extract' is the best name here as to me it does not 'extract' anything. can we use 'apply_curvature_correction' or similar?